src/configure/helper/graphHelper.ts

```diff
@@ -0,0 +1,212 @@
+import { AzureSession, Token, AadApplication } from '../model/models';
+import { AzureEnvironment } from 'ms-rest-azure';
+import { TokenResponse, MemoryCache, AuthenticationContext } from 'adal-node';
+import * as util from 'util';
+import { Messages } from '../messages';
+import { TokenCredentials, UrlBasedRequestPrepareOptions, ServiceClientCredentials } from 'ms-rest';
+import { generateRandomPassword } from './commonHelper';
+import * as Q from 'q';
+import { RestClient } from '../clients/restClient';
+const uuid = require('uuid/v1');
+
+export class GraphHelper {
+
+    private static contributorRoleId = "b24988ac-6180-42a0-ab88-20f7382dd24c";
+    private static retryCount = 20;
+    private static retryTimeout = 2 * 1000;
+
+    public static async createSpnAndAssignRole(session: AzureSession, aadAppName: string, scope: string): Promise<AadApplication> {
+        let graphCredentials = await this.getGraphToken(session);
+        let tokenCredentials = new TokenCredentials(graphCredentials.accessToken);
+        let tenantId = session.tenantId;
+        var aadApp: AadApplication;
+
+        return this.createAadApp(tokenCredentials, aadAppName, tenantId)
+        .then((aadApplication) => {
+            aadApp = aadApplication;
+            return this.createSpn(tokenCredentials, aadApp.appId, tenantId);
+        })
+        .then((spn) => {
+            aadApp.objectId = spn.objectId;
+            return this.createRoleAssignment(session.credentials, scope, aadApp.objectId);
```

Comment:
Should we not be passing TokenCredentials here ?

Reply:
Role assignment is done in ARM so its ARM call and not graph. Therefore we are passing arm credentials.